src/Logging/Logging.Testing/src/Xunit/LoggedTestInvoker.cs

```diff
@@ -51,12 +51,13 @@ protected override object CreateTestClass()
             if (testClass is LoggedTestBase loggedTestBase)
             {
                 // Used for testing
+
                 loggedTestBase.RetryContext = _retryContext;
 
                 if (_retryContext != null)
                 {
                     // Log retry attempt as warning
-                    if (_retryContext.CurrentIteration > 0)
+                    if (_retryContext.CurrentIteration > 0 && !string.IsNullOrEmpty(_retryContext.Reason))
```

Comment:
Another reason to use a different context. The reason may not be set when using the Retry attribute but we'd still want to log a warning.